r/R/dplyr-funcs-datetime.R

```diff
@@ -172,6 +172,15 @@ register_bindings_datetime_components <- function() {
     (call_binding("yday", x) - 1) %/% 7 + 1
   })
 
+  register_binding("qday", function(x) {
+    x <- build_expr("floor_temporal", x, options = list(unit = 6L, calendar_based_origin = FALSE))
+    x <- build_expr("cast", x, options = cast_options(to_type = timestamp("s")))
+    floored_x <- build_expr("floor_temporal", x, options = list(unit = 9L, calendar_based_origin = FALSE))
+    duration_between <- build_expr("subtract", x, floored_x)
+    seconds_between <- build_expr("cast", duration_between, options = cast_options(to_type = int64()))
+    build_expr("floor", seconds_between / Expression$scalar(86400L) + Expression$scalar(1L))
```

Comment:
Please can you add some comments here explaining why we need these steps?

Reply:
Will do. I've not yet given up on optimising it some too :).

Reply:
I'm starting to suspect `days_between` will work once [ARROW-16932](https://issues.apache.org/jira/browse/ARROW-16932) is in. Will test.